### ISSUE TITLE
Async arrow functions always fail arrow-parens rule

### ISSUE THREAD

**Issue by hzoo** [babel/eslint-plugin-babel#8]
From https://github.com/eslint/eslint/issues/3354#issuecomment-129878120

> Async arrow functions always fail arrow-parens rule.

```js
async () => {
}
```
Produces error:

> error  Expected parentheses around arrow function argument     arrow-parens

@leebyron

Will probably have to be done here since async isn't standard yet

**Comment by leebyron** [comment]
Thanks for moving the issue over :)

**Comment by hzoo** [comment]
Should just have to add a line to the rule to check for async (or something like that) right after https://github.com/eslint/eslint/blob/b311fcdb19552d4da7754c05849f50ed06f36442/lib/rules/arrow-parens.js#L23

`if (node.async) token = context.getTokenAfter(token);`

Forgetting why we have https://github.com/babel/babel-eslint/blob/master/acorn-to-esprima.js#L242-L245 - I'm assuming it's some compatibility thing and we probably don't have tests for that stuff.

**Comment by jquense** [comment]
ya should be an easy enough to fix to add the rule in to our collection. Not sure when I can get to it but hopefully soon :)

**Comment by hzoo** [comment]
Yeah I tried it locally with that line so just have to add the files/tests although I would have to not delete `node.async` in babel-eslint
